src/key.rs

```diff
@@ -291,6 +290,18 @@ impl PublicKey {
         ret
     }
 
+    /// Writes the key as a byte-encoded pair of values. In compressed form
```

Comment:
In compressed form it's in fact not a pair of values.

It should be documented that  `constants::PUBLIC_KEY_SIZE` bytes are written because that's not clear from the prototype. (Not relevant to this PR but I think general the docs can be improved also for the `serialize` functions. What the user should know is that it outputs x bytes but the user shouldn't care how these bytes can be interpreted.)